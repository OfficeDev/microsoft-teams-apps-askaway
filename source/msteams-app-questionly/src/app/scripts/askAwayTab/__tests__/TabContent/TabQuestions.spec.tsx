/**
 * @jest-environment jsdom
 */

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { ThemePrepared } from '@fluentui/react-northstar';
import { configure, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';
import { Helper } from '../../shared/Helper';
import { TabQuestions } from '../../TabContent/TabQuestions';
import { themeMock } from '../mocks/themes';

configure({ adapter: new enzymeAdapterReact16() });

describe('TabQuestions Component', () => {
    let onClickAction;
    let activeSessionData;
    let theme;
    let t;

    beforeAll(() => {
        t = jest.fn();
        activeSessionData = new Helper().createEmptyActiveSessionData();
        onClickAction = jest.fn();
        theme = themeMock;
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<TabQuestions t={t} theme={theme} activeSessionData={activeSessionData} onClickAction={onClickAction} teamsTabContext={{ entityId: '', locale: '' }} />);

        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<TabQuestions t={t} theme={theme} activeSessionData={activeSessionData} onClickAction={onClickAction} teamsTabContext={{ entityId: '', locale: '' }} />);
        console.log(component.debug());

        expect(component.find('div.question-container')).toBeTruthy();
    });
});
