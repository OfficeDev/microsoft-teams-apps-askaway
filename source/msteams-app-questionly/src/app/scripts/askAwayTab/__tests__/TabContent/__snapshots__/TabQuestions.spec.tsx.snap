// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabQuestions Component should match the snapshot 1`] = `
<div
  className="question-container"
/>
`;
